components/core/src/clp/ffi/KeyValuePairLogEvent.hpp

```diff
@@ -60,6 +61,16 @@ class KeyValuePairLogEvent {
 
     [[nodiscard]] auto get_utc_offset() const -> UtcOffset { return m_utc_offset; }
 
+    /**
+     * Serializes the log event into a `nlohmann::json` object.
+     * @return A result containing the serialized JSON object or an error code indicating the
+     * failure:
+     * - std::errc::protocol_error if a `nlohmann::json::exception` occurs, or typecasting a `Value`
+     *   instance fails.
```

Comment:
``suggestion
     * - std::errc::protocol_error if a value in the log event couldn't be decoded or it couldn't be
     *   inserted into a JSON object.
     * - std::errc::result_out_of_range if a node ID in the log event doesn't exist in the schema
     *   tree.
```

* Since we're not actually covering *all* the reasons for failure, I tried to write a description that covers them at a high level.
* We missed `result_out_of_range` previously.

Reply:
I think we should try to merge the error code PR so we can improve our error report